src/v/utils/retry_chain_node.h

```diff
@@ -196,6 +197,9 @@ struct retry_permit {
 /// timeout value have to be smaller that the one of the parent node.
 class retry_chain_node {
 public:
+    using milliseconds_uint16_t
+      = std::chrono::duration<uint16_t, std::chrono::milliseconds::period>;
```

Comment:
it is necessary to store the backoff as something other than the default underlying type? I assume anything specific to uint16_t is coming from configs or checking max, and that could happen at call size as a specialization?

Reply:
I think that this makes some conversions more convenient and readable. There are few places where we need to convert it to milliseconds and then to ss::lowres_clock::duration and then convert result back to milliseconds. If we will store it as an underlying type the first conversion will have to be explicit.

Reply:
> I think that this makes some conversions more convenient and readable. There are few places where we need to convert it to milliseconds and then to ss::lowres_clock::duration and then convert result back to milliseconds. If we will store it as an underlying type the first conversion will have to be explicit.

I think the question is "why not use `std::chrono::milliseconds`?"

Reply:
Oh, milliseconds is `signed integral type of at least 45 bits`. And the `retry_chain_node` is 40 bytes.

Reply:
thanks @Lazin so the reason here is related to building smaller objects in memory.